python/graphstorm/dataloading/dataset.py

```diff
@@ -787,3 +789,79 @@ def infer_idxs(self):
         """ Set of nodes to do inference.
         """
         return self._infer_idxs
+
+class GSDistillData(Dataset):
+    """ Dataset for distillation
+
+    Parameters
+    ----------
+    file_list : list of str
+        List of input files.
+    tokenizer : transformers.AutoTokenizer
+        HuggingFace Tokenizer.
+    max_seq_len : int
+        Maximum sequence length.
+    device : str
+        Device name.
+
+    """
+    def __init__(self, file_list, tokenizer, max_seq_len, device):
+        super().__init__()
+        self.file_list = file_list
+        self.tokenizer = tokenizer
+        self.max_seq_len = max_seq_len
+        self.device = device
+        self.token_id_inputs, self.labels = self.get_inputs()
+
+    def get_inputs(self):
+        """ Tokenize textual data."""
+        for i, file_name in enumerate(self.file_list):
+            if i == 0:
+                inputs = pd.read_parquet(file_name)
+            else:
+                inputs = pd.concat([inputs, pd.read_parquet(file_name)])
+
+        token_id_inputs = []
+        for i in range(len(inputs["textual_feats"])):
+            tokens = self.tokenizer.tokenize(inputs["textual_feats"][i])
+            tokens.insert(0, self.tokenizer.cls_token) # cls token for pooling
+            token_ids = self.tokenizer.convert_tokens_to_ids(tokens)
+            token_ids = token_ids[0:min(len(token_ids), self.max_seq_len)]
+            token_id_inputs.append(token_ids)
+        labels = inputs["embeddings"].to_numpy()
+
+        return token_id_inputs, labels
+
+    def __len__(self):
+        return len(self.token_id_inputs)
+
+    def __getitem__(self, index):
+        input_ids = th.tensor(self.token_id_inputs[index], dtype=th.int64, device="cpu")
+        labels = th.tensor(self.labels[index], dtype=th.float, device="cpu")
+
+        return {
+            "input_ids": input_ids,
+            "labels": labels,
+        }
+
+    def get_collate_fn(self):
+        '''get collate function
+        '''
+        def collate_fn(batch):
+            ''' Pad tensors in a batch to the same length.
+            '''
+            ## pad inputs
+            input_ids_list = [x["input_ids"] for x in batch]
+            pad_input_ids = th.nn.utils.rnn.pad_sequence(input_ids_list, 
```

Comment:
Shouldn't we use tokenizer's padding function?

Reply:
Tokenizer's padding only takes str input. However, the input_ids_list here is a list of tensor. So i use th.nn.utils.rnn.pad_sequence, which has been a common padding function since pre-Transformer years.